### PR TITLE
Add utility scene 'Node' to CreateRootDialog

### PR DESCRIPTION
**Pull request by willnationsdev**
I would like to also be able to create simple Node scenes from the CreateRootDialog (which I really love btw). I create empty utility scenes ALL the time.

create_root_dialog

Edit: I renamed it from "Utility Scene" to "Utility" since I think that looks cleaner / more sensible to someone who is new to the engine.

### CODE REVIEW

**Comment by groud** [comment]
Utility is a quite a bad name I think. But the idea is good.

**Comment by akien-mga** [comment]
> Utility is a quite a bad name I think.

Agreed, or at least it should come *after* Control, not in the first position. The purpose of this feature is discoverability for new users, and new users won't start learning Godot with an empty Node. They need 2D, 3D or Control in priority.

**Comment by Overblob** [comment]
It could be named something like "General Purpose Scene" or "General Purpose Node" (or even "Multi-Purpose Node"), although it's maybe a bit too long.

But something in that spirit, without categorizing too much the basic Node class.

**Comment by nunodonato** [comment]
how about "Misc"? just because "general purpose" is too long :p 

@willnationsdev  a bit offtopic but, what do you use these kind of scenes for?

**Comment by Overblob** [comment]
Or maybe "Generic Node" or "Generic Scene" then? Generic as opposed à specific

**Comment by willnationsdev** [comment]
I like "Generic".

@nunodonato I may be creating scenes that wrap several audio players, or a scene that holds a hierarchy of data-as-nodes provided by a plugin, or I could just be protoyping something in script code and just want to throw a plain old node into the root.

**Comment by willnationsdev** [comment]
Here is an updated image for folks on the latest commit amend.

create_root1

**Comment by Overblob** [comment]
I personnally find "Generic" a bit too short, visually speaking

**Comment by willnationsdev** [comment]
@Overblob I could always change it to "Genericalicity Scene Node".

**Comment by Overblob** [comment]
😃

@willnationsdev : Actually, my remark was rather to point out that other entries in this list was named as "Adjective + **Name**", like in "2D **Scene**" or "Custom **Node**"

So for uniformity, I'd rather have seen "Generic **Scene**" or "Generic **Node**".

Well, I admit I can sometimes be a little ridig stickler when it comes to homogenization, but I've really no problem accepting other's choices as quality may come from exchanges and consensus.

So if it's "Generic", be it!

**Comment by willnationsdev** [comment]
I mean, I'd be fine with "Generic Scene" too. What do you guys all think @akien-mga @Zephilinox @nunodonato ?

**Comment by nunodonato** [comment]
I like "Generic" more than "Generic Scene", but both are ok :)
(in the same way that you didn't write "user interface scene")

**Comment by willnationsdev** [comment]
Ok, I think I'm just gonna leave it as Generic. This PR should be all good to go @mhilbrunner .

**Comment by fire** [comment]
I agree with @Overblob 's arguments.  "Generic Scene" does fit in better.

**Comment by reduz** [comment]
I vote closing this. The main goal of this is dialog is that it should be for new users. Creating an empty node is not even close to being a common case for them. Whatever name you add will be confusing, so i think it's just not good.

**Comment by LikeLakers2** [comment]
We're starting a vote? Over a utility button? I vote to merge because I fail to see how this will be confusing to any new user. We already have "2D Scene", "3D Scene", and "User Interface" which are much more descriptive -- and I'm fairly sure even new people will think "Hey! This option will probably serve my needs better than that generic option."

**Comment by akien-mga** [comment]
I don't think it will be notably confusing to new users... but I don't think it serves any purpose either. The 2D Scene, 3D Scene and UI buttons *do* serve a purpose of discoverability for new users, to let them start with a root node corresponding to what they want to do.

For all other, more advanced use cases, these buttons are not useful. If I want a Sprite, a KinematicBody, a CanvasLayer, etc., I still need to use the normal way of adding new nodes via the Create New Node dialog. I don't think starting a scene with a base Node is such a common use case that it needs to be given one-click access like those which were added for discoverability.

In short: This feature is to make Godot more user-friendly to beginners. It's not to save one click to power users. If we merge this, we'll soon have requests to add more "one-click instance" node types, or even a one click access to favorites/custom node types, etc., which is way out of the scope of this feature.

**Comment by groud** [comment]
That being said, I think a good solution might be to display the nodes marked as favorite there. It would make the feature more useful that it is now.

**Comment by willnationsdev** [comment]
@groud I agree. I definitely think reduz/akien have a point in regards to the discoverability aspects, but I also think using this space more effectively to quick-start a scene is a good idea, especially for non-beginners who wanna move quickly. Maybe create an HSeparator and then having a list of the favorite nodes?

**Comment by groud** [comment]
For me, the best UX would be to have the default 2d/3d/gui nodes by default. Thed they would be completely replaced if you have at least one favored node.

This keeps the discoverability feature for new users but allow a full customization for power users.

**Comment by LikeLakers2** [comment]
I'm not sure I'd agree with replacing it when they have one favored node. Perhaps keep the 2D/3D/GUI buttons there, and have a button labeled Favorite nodes if they have at least one? You could use a menu there that automatically populates itself with the user's favorite nodes.

On a similar note, and I don't know if this the appropriate place to put this, but I've been feeling the name "Custom Node" is confusing considering Custom Nodes are usually just nodes that... well, are added by addons.

----

@akien-mga That was not the impression I got from his message at first, but I'll chalk that down to my own way of understanding. Regardless, thanks for clarifying. I hope my message didn't come off as too rude, or at the least, that I haven't made reduz uncomfortable here. And just to clarify how I saw it, I was seeing what looked like some vague "It's confusing" statement with no explanation as to why it's confusing, and I feel the need to say something about it.

**Comment by akien-mga** [comment]
Given the above discussion, I think there's no consensus for adding a "Generic" button to one-click instantiate a Node in this Create Root Node panel.

What would be more interesting IMO based on the above discussion is a button (whose state would be saved to project metadata) to fully replace the Create Root Node panel by another one that would only include favorite nodes to one-click instantiate as root node. This would be more useful for power users, while safe-guarding the discoverability of the base node types for 2D, 3D and GUI.

**Comment by Xrayez** [comment]
> In short: This feature is to make Godot more user-friendly to beginners. It's not to save one click to power users. If we merge this, we'll soon have requests to add more "one-click instance" node types, or even a one click access to favorites/custom node types, etc., which is way out of the scope of this feature.

No, if this could be merged, than it would fill a void, because `Node` is so fundamental to the engine. I totally agree with the use cases described by @willnationsdev.

I've unintentionally created a "duplicate" proposal in godotengine/godot-proposals#1206 before I stumbled upon this PR, with some my own rationale regarding this feature, and how it could be possibly integrated better as to not hurt the beginner experience.
